examples/tide/main.rs

```diff
@@ -0,0 +1,186 @@
+}
+
+async fn register(mut request: tide::Request<AppState>) -> tide::Result {
+    let username = request.param("username")?.parse()?;
+    let reg = request.body_json::<RegisterPublicKeyCredential>().await?;
+    let actor_res = request.state().register(&username, &reg).await;
+    let res = match actor_res {
+        Ok(()) => tide::Response::new(tide::StatusCode::Ok),
+        Err(e) => {
+            tide::log::debug!("register -> {:?}", e);
+            tide::Response::new(tide::StatusCode::InternalServerError)
+        }
+    };
+    Ok(res)
+}
+
+async fn login(mut request: tide::Request<AppState>) -> tide::Result {
+    let username: String = request.param("username")?.parse()?;
+    let username_copy = username.clone();
+    let lgn = request.body_json::<PublicKeyCredential>().await?;
+
+    match request.state().authenticate(&username_copy, &lgn).await {
+        Ok(()) => (),
+        Err(e) => {
+            tide::log::debug!("login -> {:?}", e);
+            return Ok(tide::Response::new(tide::StatusCode::InternalServerError));
+        }
+    };
+
+    let session = request.session_mut();
+
+    // Clear the anonymous flag
+    session.remove("anonymous");
+    tide::log::debug!("removed anonymous flag");
+
+    // Set the userid
+    session.insert_raw("userid", username);
+
+    Ok(tide::Response::new(tide::StatusCode::Ok))
+}
+
+#[async_std::main]
+async fn main() -> tide::Result<()> {
+    let opt: CmdOptions = CmdOptions::from_args();
+
+    if opt.debug {
+        tide::log::with_level(tide::log::LevelFilter::Debug);
+    } else {
+        tide::log::start();
+    }
+
+    tide::log::debug!("Started logging ...");
+
+    let prefix = opt.prefix.clone();
+    let domain = opt.rp_id.clone();
+
+    let wan_c = WebauthnEphemeralConfig::new(
+        opt.rp_name.as_str(),
+        opt.rp_origin.as_str(),
+        opt.rp_id.as_str(),
+        None,
+    );
+
+    let wan = WebauthnActor::new(wan_c);
+
+    let app_state = Arc::new(wan);
+
+    let mut app = tide::with_state(app_state);
+    let cookie_sig = StdRng::from_entropy().gen::<[u8; 32]>();
+    let sessions =
+        tide::sessions::SessionMiddleware::new(tide::sessions::MemoryStore::new(), &cookie_sig)
+            .with_cookie_path(prefix.as_str())
+            .with_cookie_domain(domain.as_str())
+            .with_same_site_policy(tide::http::cookies::SameSite::Strict)
+            .with_cookie_name("webauthnrs");
+    app.with(sessions);
+    app.with(tide::log::LogMiddleware::new());
+    app.at("/auth").get(index_view);
+    app.at("/auth/challenge/register/:username")
+        .post(challenge_register);
+    app.at("/auth/challenge/login/:username")
+        .post(challenge_login);
+    app.at("/auth/register/:username").post(register);
+    app.at("/auth/login/:username").post(login);
+    app.at("/auth/static/").serve_dir("static")?;
+
+    // TODO: tide_rustls uses version 0.14.0 (we're on 0.15.0) and uses rustls over openssl
+    // if opt.enable_tls {
```

Comment:
See above comment about tls

Reply:
Just use tide 0.14 then :)

Reply:
Oh this is fun. Tide 0.14 doesn't have `with_cookie_domain` on the `SessionMiddleware` object.

Reply:
For now, we can just remove it. it's not the most critical thing in this application.

Reply:
Ah it looks like tide-rustls got updated to the correct version since I PRd.